### PR TITLE
Backport fixes

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #5468 #5519 #5520 #5580 #5577 #5562 #5599

Backport of latest security patches on 9.3 branch.

### CODE REVIEW

File: install/update_931_932.php [hunk 1]
```diff
@@ -73,6 +73,12 @@ function update931to932() {
    $migration->addKey('glpi_states', 'is_visible_enclosure');
    $migration->addKey('glpi_states', 'is_visible_pdu');
 
+   // Create a dedicated token for rememberme process
+   if (!$DB->fieldExists('glpi_users', 'cookie_token')) {
+      $migration->addField('glpi_users', 'cookie_token', 'string', ['after' => 'api_token_date']);
+      $migration->addField('glpi_users', 'cookie_token_date', 'datetime', ['after' => 'cookie_token']);
+   }
+
```
Comment (remicollet) [inline install/update_931_932.php]: Either you add this in a new file, or you don't apply the fix.
This will never be run.

Comment (trasher) [reply install/update_931_932.php]: This can be using the cli --force option

Comment (cedric-anne) [reply install/update_931_932.php]: Creating another file will create a mess for user who will try to update to GLPI 9.4.0 or 9.4.1 in a second time.
I would prefer to add a clear message on the release message to indicate that the update command have to be launched using CLI and maybe give DB queries to execute for people who cannot use CLI.

**Comment by cedric-anne** [comment]
You'll have to get #5612 and #5599 fixes.

File: inc/auth.class.php [hunk 2]
```diff
@@ -502,7 +502,7 @@ function getAlternateAuthSystemsUserLogin($authtype = 0) {
 
                   $token = User::getToken($cookie_id, 'personal_token');
 
-                  if ($token !== false && Auth::checkPassword($token, $cookie_token)) {
+                  if (Auth::checkPassword($cookie_token, $hash)) {
```
Comment (ddurieux) [inline inc/auth.class.php]: in 9.3, it's:
```
if (Auth::checkPassword($cookie_token, $token)) {
```

Comment (trasher) [reply inc/auth.class.php]: You're right, thank you. I also saw that `personal_token` was still used instead of `cookie_token`.

**Comment by remicollet** [comment]
I definitively think we need a 9.3.4, with an update script.

File: inc/document.class.php [hunk 3]
```diff
@@ -1497,8 +1497,17 @@ static function getImageTag($string) {
     * @return boolean
     */
    public static function isImage($file) {
-      $ext = strtolower(pathinfo($file, PATHINFO_EXTENSION));
-      return (in_array($ext, ['jpg', 'jpeg', 'png', 'bmp', 'gif']));
+      if (extension_loaded('exif')) {
+         $etype = exif_imagetype($file);
```
Comment (ddurieux) [inline inc/document.class.php]: I have an error:
```
 PHP Warning(2): exif_imagetype(_pictures/3a/148_599593015b13a_min.png): failed to open stream: No such file or directory
```

File: install/update_931_932.php (hunk 1 above)
Comment (trasher) [reply install/update_931_932.php]: DB change has been dropped

**Comment by cedric-anne** [comment]
I've push some modifications:
- fix of `Fix chevrons display in select2 rendering` commit (was not fully picked),
- removal of `Fix search of tickets_id in mail headers; fixes #5625` commit (not related to security fixes and not working as it on 9.3 branch),
- addition of `Fix escaping of optgroups in dropdowns; fixes #5646`.

**Pull request merged**